src/expr.rs

```diff
@@ -1909,6 +1909,60 @@ pub mod parsing {
+            impl Parse for $expr_type {
+                fn parse(input: ParseStream) -> Result<Self> {
+                    let expr: Expr = input.parse()?;
+                    match expr {
+                        Expr::$variant(inner) => Ok(inner),
+                        _ => Err(Error::new_spanned(expr, $msg))
+                    }
+                }
+            }
+        )
+    }
+
+    impl_trivial_parse!(ExprBox, Box, "expected box expression");
+    impl_trivial_parse!(ExprInPlace, InPlace, "expected placement expression");
+    impl_trivial_parse!(ExprArray, Array, "expected slice literal expression");
+    impl_trivial_parse!(ExprCall, Call, "expected function call expression");
+    impl_trivial_parse!(ExprMethodCall, MethodCall, "expected method call expression");
+    impl_trivial_parse!(ExprTuple, Tuple, "expected tuple expression");
+    impl_trivial_parse!(ExprBinary, Binary, "expected binary operation");
+    impl_trivial_parse!(ExprUnary, Unary, "expected unary operation");
+    //impl_trivial_parse!(ExprLit, Lit, "expected literal in place of an expression");
+    impl_trivial_parse!(ExprCast, Cast, "expected cast expression");
+    impl_trivial_parse!(ExprType, Type, "expected type ascription expression");
+    impl_trivial_parse!(ExprLet, Let, "expected let guard");
+    //impl_trivial_parse!(ExprIf, If, "expected if expression");
+    //impl_trivial_parse!(ExprWhile, While, "expected while loop");
+    //impl_trivial_parse!(ExprForLoop, ForLoop, "expected for loop");
+    //impl_trivial_parse!(ExprLoop, Loop, "expected conditionless loop");
+    impl_trivial_parse!(ExprClosure, Closure, "expected closure expression");
+    impl_trivial_parse!(ExprUnsafe, Unsafe, "expected unsafe block");
+    impl_trivial_parse!(ExprBlock, Block, "expected blocked scope");
+    impl_trivial_parse!(ExprAssign, Assign, "expected assignment expression");
+    impl_trivial_parse!(ExprAssignOp, AssignOp, "expected compound assignment expression");
+    impl_trivial_parse!(ExprField, Field, "expected struct field access");
+    impl_trivial_parse!(ExprIndex, Index, "expected indexing expression");
+    impl_trivial_parse!(ExprRange, Range, "expected range expression");
+    impl_trivial_parse!(ExprReference, Reference, "expected referencing operation");
+    impl_trivial_parse!(ExprBreak, Break, "expected break expression");
+    impl_trivial_parse!(ExprContinue, Continue, "expected continue expression");
+    impl_trivial_parse!(ExprReturn, Return, "expected return expression");
+    impl_trivial_parse!(ExprMacro, Macro, "expected macro invocation expression");
+    impl_trivial_parse!(ExprStruct, Struct, "expected struct literal expression");
+    impl_trivial_parse!(ExprRepeat, Repeat, "expected array literal constructed from one repeated element");
+    impl_trivial_parse!(ExprParen, Paren, "expected parenthesized expression");
+    impl_trivial_parse!(ExprGroup, Group, "expected ExprGroup");
```

Comment:
ExprGroup needs to be handled in the match that you generate for each variant and should not have its own Parse impl. Groups are generated by the macro system to control precedence around metavariables -- imagine them like invisible parens. In your range example, we would want the ExprRange parser to support `#[attr($range, ...)]` where $range is a range expression.